clang/lib/Driver/Driver.cpp

```diff
@@ -5399,14 +5416,26 @@ InputInfo Driver::BuildJobsForActionNoCache(
                                                 DependentOffloadKind)}] =
           CurI;
     }
-
-    // Now that we have all the results generated, select the one that should be
-    // returned for the current depending action.
-    std::pair<const Action *, std::string> ActionTC = {
-        A, GetTriplePlusArchString(TC, BoundArch, TargetDeviceOffloadKind)};
-    assert(CachedResults.find(ActionTC) != CachedResults.end() &&
-           "Result does not exist??");
-    Result = CachedResults[ActionTC];
+    // Do a check for a dependency file unbundle for FPGA.  This is out of line
+    // from a regular unbundle, so just create and return the name of the
+    // unbundled file.
+    if (JA->getType() == types::TY_FPGA_Dependencies) {
+      // Unbundling of dependency file for FPGA is target only
```

Comment:
```suggestion
```
This comment is a bit confusing. Does "target" imply "device compilation" here? Also, it doesn't seem to tie in well with the surrounding code.

Reply:
It's meant to imply there is only one output file generated, I'll remove it as the general comment above states the purpose already.